tests/testsuite/deduplicate_workspace.rs

```diff
@@ -0,0 +1,861 @@
+        .feature("testing", &vec![])
+        .publish();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo()
+        .with_stderr(
+            "\
+[UPDATING] `[..]` index
+[DOWNLOADING] crates ...
+[DOWNLOADED] dep v0.1.2 ([..])
+[COMPILING] dep v0.1.2
+[COMPILING] bar v0.2.0 ([CWD])
+[FINISHED] dev [unoptimized + debuginfo] target(s) in [..]
+",
+        )
+        .run();
+
+    p.cargo("check").masquerade_as_nightly_cargo().run();
+    let lockfile = p.read_lockfile();
+    assert!(lockfile.contains("dep"));
+    p.cargo("publish --token sekrit")
+        .masquerade_as_nightly_cargo()
+        .run();
+    publish::validate_upload_with_contents(
+        r#"
+        {
+          "authors": [],
+          "badges": {},
+          "categories": [],
+          "deps": [
+            {
+              "default_features": true,
+              "features": ["testing"],
+              "kind": "normal",
+              "name": "dep",
+              "optional": false,
+              "registry": "https://github.com/rust-lang/crates.io-index",
+              "target": null,
+              "version_req": "^0.1.2"
+            }
+          ],
+          "description": null,
+          "documentation": null,
+          "features": {},
+          "homepage": null,
+          "keywords": [],
+          "license": null,
+          "license_file": null,
+          "links": null,
+          "name": "bar",
+          "readme": null,
+          "readme_file": null,
+          "repository": null,
+          "vers": "0.2.0"
+          }
+        "#,
+        "bar-0.2.0.crate",
+        &["Cargo.toml", "Cargo.toml.orig", "Cargo.lock", "src/main.rs"],
+        &[(
+            "Cargo.toml",
+            &format!(
+                r#"{}
+cargo-features = ["workspace-inheritance"]
+
+[package]
+name = "bar"
+version = "0.2.0"
+authors = []
+
+[dependencies.dep]
+version = "0.1.2"
+features = ["testing"]
+"#,
+                cargo::core::package::MANIFEST_PREAMBLE
+            ),
+        )],
+    );
+}
+
+#[cargo_test]
+fn inherit_from_own_undefined_field() {
+    registry::init();
+
+    let p = project().build();
+
+    let _ = git::repo(&paths::root().join("foo"))
+        .file(
+            "Cargo.toml",
+            r#"
+            cargo-features = ["workspace-inheritance"]
+
+            [package]
+            name = "foo"
+            version = "1.2.5"
+            authors = ["rustaceans"]
+            description = { workspace = true }
+
+            [workspace]
+            members = []
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build")
+        .masquerade_as_nightly_cargo()
+        .with_status(101)
+        .with_stderr(
+            "\
+[ERROR] failed to parse manifest at `[CWD]/Cargo.toml`
+
+Caused by:
+  error reading `description` from workspace root manifest's `[workspace.description]`
+
+Caused by:
+  [workspace.description] was not defined
```

Comment:
I think I'd change this to
```
`workspace.description` was not defined
```
If we could say in what file its not defined, that'd be even better.  It should be obvious but it doesn't help.

Reply:
I'll change it to this!